src/System.ServiceModel.Federation/src/System/ServiceModel/Federation/WSTrustChannelSecurityTokenProvider.cs

```diff
@@ -167,6 +167,16 @@ protected virtual WsTrustRequest CreateWsTrustRequest()
                 trustRequest.ComputedKeyAlgorithm = _requestSerializationContext.TrustKeyTypes.PSHA1;
             }
 
+            if (WSTrustTokenParameters.Claims != null)
+            {
+                trustRequest.Claims = WSTrustTokenParameters.Claims;
+            }
```

Comment:
You need to copy the values over. If someone override this method and modifies the Claims collection, they will end up modifying WSTrustTokenParameters.Claims too.

Reply:
Updated for Claims and AdditionalRequestParameters, could you please help double confirm?

Reply:
Hi @mconnew, could you please help double confirm if the updated code looks good to you? Thanks!